docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,329 @@
+
+```go
+// GetCAR takes DataRoot and returns a buffered reader to respective EDS serialized as CARv1 file.
+// 
+// The Reader strictly reads the first quadrant(1/4) of EDS omitting all the NMT Merkle proofs.
+// Integrity of the store data is not verified. 
+// 
+// Caller must Close returned reader after reading.
+func (s *Store) GetCAR(context.Context, DataRoot) (io.ReadCloser, error)
+```
+
+##### `share.EDSStore.Blockstore`
+
+`Blockstore` method return a `Blockstore` interface implementation instance, providing random access over share and NMT
+Merkle proof in every stored EDS. It is required for FNs/BNs to serve DAS requests over the Bitswap and for reading data
+by namespace. // TODO Link to the section
+
+There is a [frozen/un-merged implementation](https://github.com/filecoin-project/dagstore/pull/116) of `Blockstore`
+over `DAGStore` and CARv2 indexes.
+
+NOTE: We can either use it(and finish if something is missing for our case) or implement custom optimized for our needs.
+
+```go
+// Blockstore returns an IPFS Blockstore providing access to individual shares/nodes of all EDS 
+// registered on the Store. NOTE: The Blockstore does not store full Celestia Blocks, but IPFS blocks. 
+// We represent `shares` and NMT Merkle proofs as IPFS blocks and IPLD nodes, so that Bitswap can access those.
+func (s *Store) Blockstore() blockstore.Blockstore
+
+```
+
+##### `share.EDSStore.Get`
+
+To read an entire EDS `Get` method is introduced. Internally it:
+
+- Gets serialized EDS `io.Reader` via `Store.GetCAR`
+- Deserializes EDS and validates it via `share.ReadEDS`
+
+NOTE: It's not necessary, but an API ergonomics/symmetry nice-to-have
+
+```go
+// Get reads EDS out of Store by given DataRoot.
+// 
+// It reads only one quadrant(1/4) of the EDS and verifies integrity of the stored data by recomputing it.
+func (s *Store) Get(context.Context, DataRoot) (*rsmt2d.ExtendedDataSquare, error)
+
+```
+
+##### `share.EDSStore.Remove`
+
+To remove stored EDS `Remove` methods is introduced. Internally it:
+
+- Converts `DataRoot` into the [`shard.Key`](<https://github.com/filecoin-project/dagstore/blob/master/shard/key.go#L12>
+- Removes respective `Mount` and `Shard` from `DAGStore`
+  - `DAGStore` cleans up indices and `mount.Registery` itself
+- Removes `FileMount` file of CARv1 file from disk under `storepath/DataRoot` path
+
+NOTES:
+
+- It's not necessary, but an API ergonomics/symmetry nice-to-have
+- GC logic on the DAGStore has to be investigated so that Removing is correctly implemented
+
+```go
+// Remove removes EDS from Store by given DataRoot and cleans up all the indexing.
+func (s *Store) Remove(context.Context, DataRoot) error
+
+```
+
+#### Reading Data By Namespace
+
+Generally stays unchanged with minor edits:
+
+- `share/ipld.GetByNamespace` is kept to load data from disk only and not from network anymore
+  - Using `Blockservice` with offline exchange // TODO Links
```

Comment:
We can enable this, but the case where it is missing something locally should never happen, otherwise, it does not store the block. 

The protocol above will respond as not found in this case, and the requesting client will go ask someone else. This is a better strategy than waiting for the Server to find somewhere else data it does not have.